catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -810,6 +878,41 @@ public CreateResponse create(CreateRequest createRequest)
 
         Exception ingestError = null;
         try {
+            if (policy != null) {
```

Comment:
Can we extract private methods for the policy and access checks to keep `create()` shorter and self documenting? And maybe make them so that it can be re-use by the other method as well?

I would actually go as far as suggesting that we create a separate class to handle all the access and policy stuff instead of stuffing this all in here and delegate to it but I already know what the answer is going to be :-) Would help not grow this class any bigger and make this new code unit testable though ;-)